### DISCUSSION TITLE
Return response from Orchestrator Generator to display in Mia

### DISCUSSION THREAD

**Discussion by rmuller-vu** [mia-platform/community#404] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

1.95.2-hotfix.0

### Description

We are wanting to return a message if something goes wrong with the generation of the orchestrator files. Is there a way we can return a custom message to Mia from the Orchestrator?

### Expected Outcome

Generation fails and the user sees a red rocket ship on the deploy page in the console with a custom message sent from the Orchestrator.

**Comment by fredmaggiowski** [comment]
Hi @rmuller-vu, thanks for reaching out! 

I think that's totally feasible and should be easily included in one of the upcoming releases, let me track it down in our backlog and come back to you later with the response a response payload definition that makes sense

**Reply by fredmaggiowski** [reply]
Hey @rmuller-vu, I though a little bit about the response payload your generator should provide and came up with the following one:

```json
{
   "error": "some technical error you can trace in service logs", // having access to Console service logs, this may be helpful to trace errors causes
   "message": "some user facing error message that is propagated to the final user" 
}
```

does it make sense to you?

**Reply by rmuller-vu** [reply]
This looks good and fits our needs. Would you have a rough estimate when this feature would be available?

**Reply by fredmaggiowski** [reply, marked as answer]
I had some time this morning and since it was a pretty straightforward change I implemented it just in time to have it featured in next release v12.2.2 which will be GA in two weeks (a beta version will be available in a few days but I'd not recommend installing it unless you have a self-hosted testing console installation)

The final result should look like this, where the _"this message should be seen by the user"_ is the message returned by my mock generator.

<img width="663" alt="image" src="https://github.com/mia-platform/community/assets/7142570/272e036b-f4d6-44af-982a-005e41cbede7">

---

Here is the PR for the updated documentation that will be released with v12.2.2: https://github.com/mia-platform/documentation/pull/1194
